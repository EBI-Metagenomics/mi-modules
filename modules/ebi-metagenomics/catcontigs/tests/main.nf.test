
nextflow_process {

    name "Test Process CATCONTIGS"
    script "../main.nf"
    process "CATCONTIGS"

    tag "modules"
    tag "modules_ebimetagenomics"
    tag "catcontigs"

    test("bins-fasta - cat taxonomy") {

        when {
            process {
                """
                input[0] = [
                    [ id:'test', single_end:false ], // meta map
                    file("${moduleDir}/tests/fixtures/contigs.fasta", checkIfExists: true)
                ]
                input[1] = file("${moduleDir}/tests/fixtures/cat_db/db", checkIfExists: true)
                input[2] = file("${moduleDir}/tests/fixtures/cat_db/tax", checkIfExists: true)
                """
            }
        }

        then {
            assertAll(
                { assert process.success },
                { assert snapshot(process.out).match() }
            )
        }

    }

    test("bins-fasta - cat taxonomy - stub") {

        options "-stub"

        when {
            process {
                """
                input[0] = [
                    [ id:'test', single_end:false ], // meta map
                    file("${moduleDir}/tests/fixtures/contigs.fasta", checkIfExists: true)
                ]
                input[1] = file("${moduleDir}/tests/fixtures/cat_db/db", checkIfExists: true)
                input[2] = file("${moduleDir}/tests/fixtures/cat_db/tax", checkIfExists: true)
                """
            }
        }

        then {
            assertAll(
                { assert process.success },
                { assert snapshot(process.out).match() }
            )
        }

    }

}
