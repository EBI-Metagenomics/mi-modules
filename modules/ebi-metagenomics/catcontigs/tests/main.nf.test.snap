{
    "bins-fasta - cat taxonomy - stub": {
        "content": [
            {
                "0": [
                    [
                        {
                            "id": "test",
                            "single_end": false
                        },
                        "test_summary.txt:md5,d41d8cd98f00b204e9800998ecf8427e"
                    ]
                ],
                "1": [
                    [
                        {
                            "id": "test",
                            "single_end": false
                        },
                        "test_contig2classification_official_names.tsv:md5,d41d8cd98f00b204e9800998ecf8427e"
                    ]
                ],
                "2": [
                    "versions.yml:md5,b7b6ee68280d52f554f92b37e972984d"
                ],
                "contigs_classification": [
                    [
                        {
                            "id": "test",
                            "single_end": false
                        },
                        "test_contig2classification_official_names.tsv:md5,d41d8cd98f00b204e9800998ecf8427e"
                    ]
                ],
                "summary": [
                    [
                        {
                            "id": "test",
                            "single_end": false
                        },
                        "test_summary.txt:md5,d41d8cd98f00b204e9800998ecf8427e"
                    ]
                ],
                "versions": [
                    "versions.yml:md5,b7b6ee68280d52f554f92b37e972984d"
                ]
            }
        ],
        "meta": {
            "nf-test": "0.9.0",
            "nextflow": "24.04.4"
        },
        "timestamp": "2024-10-18T11:20:25.510416"
    },
    "bins-fasta - cat taxonomy": {
        "content": [
            {
                "0": [
                    [
                        {
                            "id": "test",
                            "single_end": false
                        },
                        "test_summary.txt:md5,f7eed0869f97e803c22a1a4f586aa7c9"
                    ]
                ],
                "1": [
                    [
                        {
                            "id": "test",
                            "single_end": false
                        },
                        "test_contig2classification_official_names.tsv:md5,1efb34f5b59ea561115ce262277d1a37"
                    ]
                ],
                "2": [
                    "versions.yml:md5,b7b6ee68280d52f554f92b37e972984d"
                ],
                "contigs_classification": [
                    [
                        {
                            "id": "test",
                            "single_end": false
                        },
                        "test_contig2classification_official_names.tsv:md5,1efb34f5b59ea561115ce262277d1a37"
                    ]
                ],
                "summary": [
                    [
                        {
                            "id": "test",
                            "single_end": false
                        },
                        "test_summary.txt:md5,f7eed0869f97e803c22a1a4f586aa7c9"
                    ]
                ],
                "versions": [
                    "versions.yml:md5,b7b6ee68280d52f554f92b37e972984d"
                ]
            }
        ],
        "meta": {
            "nf-test": "0.9.0",
            "nextflow": "24.04.4"
        },
        "timestamp": "2024-10-18T11:20:19.529342"
    }
}